src/js/client/editprefset-toc-widget.js

```diff
@@ -0,0 +1,49 @@
+/*
+ * editprefset-toc-widget
+ *
+ * A widget to outline the various portions of the ppt editing page, from safe
+ * overview through generic preferences, and a listing of all the applications
+ * currently added to the site.  Meant to appear on the left hand side as a sort
+ * of table of contents.
+ */
+"use strict";
+
+var gpii  = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.devpmt");
+fluid.registerNamespace("gpii.devpmt.editPrefsTocWidget");
+
+fluid.defaults("gpii.devpmt.editPrefsTocWidget", {
+    gradeNames: ["gpii.devpmt.viewComponent"],
+    model: {
+        npsetApplications: null,
+        allSolutions: null,
+        renderList: null
+    },
+    selectors: {
+        editPrefsTocContainer: "#pmt-editprefstoc-container"
+    },
+    templates: {
+        initial: "editprefset-toc-widget"
+    },
+    modelListeners: {
+        "npsetApplications": {
+            funcName: "gpii.devpmt.editPrefsTocWidget.generateRenderList",
+            args: ["{that}"]
+        },
+        "renderList": {
+            func: "{that}.reRender"
+        }
+    }
+});
+
+
+gpii.devpmt.editPrefsTocWidget.generateRenderList = function (that) {
```

Comment:
This can be achieved by a model relay from the source model path "npsetApplications" to the target "renderList".

Reply:
It required restructing the model a bit to be able to use a simple transform function (rather than building out a complex multipleInput transform), but the end result is pretty cool.